### PR TITLE
Add tests for Annex B `Object.prototype` methods

### PR DESCRIPTION
**Pull request by jugglinmike**
This patch has a large number of files, but there is a good deal of symmetry in
the tests (just like the specification text they exercise).

### CODE REVIEW

**Comment by leobalter** [comment]
You're 3 lines of code close to get it l33t.

**Comment by jugglinmike** [comment]
Let's find three assertions that I should add :P

File: test/annexB/built-ins/Object/prototype/__defineGetter__/define-non-extensible.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+description: Behavior when "this" value is not extensible
+info: |
+    [...]
+    5. Perform ? DefinePropertyOrThrow(O, key, desc).
+---*/
+
+var noop = function() {};
+var thrower = function() {
+  throw new Test262Error();
+};
```
Comment (leobalter) [inline test/annexB/built-ins/Object/prototype/__defineGetter__/define-non-extensible.js]: this is defined but not used

File: test/annexB/built-ins/Object/prototype/__defineSetter__/define-non-extensible.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+description: Behavior when "this" value is not extensible
+info: |
+    [...]
+    5. Perform ? DefinePropertyOrThrow(O, key, desc).
+---*/
+
+var noop = function() {};
+var thrower = function() {
+  throw new Test262Error();
+};
```
Comment (leobalter) [inline test/annexB/built-ins/Object/prototype/__defineSetter__/define-non-extensible.js]: defined but not used as well

**Comment by leobalter** [comment]
Instead of finding 3 extra lines we are probably getting rid of 6.

**Comment by jugglinmike** [comment]
I somehow missed this; sorry for the delay. We'll get 1337 next time

**Pull request merged**